### PR TITLE
Add devtoolset-8-{perftools,toolchain} sources

### PR DESCRIPTION
**Pull request by pkubatrh**

### CODE REVIEW

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
@mpolacek Hi Marek, can you check why the tests are failing for devtoolset-8-perftools? I suspect its something to do with how the `test_sanity_ocount_run` changed between versions.

In devtoolset-7-perftools the test is skipped when a specific message is found in the output:
https://github.com/sclorg/devtoolset-container/blob/master/7-perftools/test/run#L102

but for devtoolset-8-perftools this happens only when the architecture is s390x:
https://github.com/pkubatrh/devtoolset-container/blob/add_devtoolset8/8-perftools/test/run#L124

Is this correct? Or should there be two if clauses, one for each skip?

**Comment by hhorak** [comment]
I think this specific check might be influenced not only by different architectures, but also by different settings of the kernel, and I can imagine this check might never work correctly in our CI tests. I've checked that this upstream test is different from what we have downstream even for devtoolset-7 and since we don't pull code from github to downstream for these devtoolset images, my opinion is that we can have the difference for devtoolset-8 as well.

**Comment by hhorak** [comment]
[test]

**Comment by pkubatrh** [comment]
I did not notice that the tests are different even for 7-perftools, so makes sense I guess to include the same changes even for 8-perftools.

Done so in the latest commit, lets see if tests pass now

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
[test]

**Comment by pkubatrh** [comment]
[test]

File: 8-perftools/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+      io.k8s.display-name="Developer Toolset 8 Performance Tools" \
+      io.k8s.min-memory="2Gi" \
+      maintainer="Frank Eigler <fche@redhat.com>"
+
+RUN yum install -y centos-release-scl-rh && \
+    INSTALL_PKGS="devtoolset-8-systemtap devtoolset-8-valgrind devtoolset-8-elfutils devtoolset-8-elfutils-devel devtoolset-8-oprofile devtoolset-8-oprofile-jit devtoolset-8-oprofile-devel devtoolset-8-ltrace devtoolset-8-dwz devtoolset-8-make devtoolset-8-memstomp devtoolset-8-strace" && \
+    [ $(uname -m) = "x86_64" ] && INSTALL_PKGS="$INSTALL_PKGS devtoolset-8-dyninst devtoolset-8-dyninst-devel" || : && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    yum -y clean all --enablerepo='*'
```
Comment (phracek) [inline 8-perftools/Dockerfile:22]:
```suggestion
    yum clean all -y
```

File: 8-perftools/Dockerfile.rhel7 [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+# https://access.redhat.com/solutions/1443553
+RUN yum repolist > /dev/null && \
+    yum-config-manager --enable rhel-server-rhscl-7-rpms && \
+    yum-config-manager --enable rhel-7-server-optional-rpms && \
+    yum-config-manager --disable epel >/dev/null || : && \
+    INSTALL_PKGS="devtoolset-8-systemtap devtoolset-8-valgrind devtoolset-8-elfutils devtoolset-8-elfutils-devel devtoolset-8-oprofile devtoolset-8-oprofile-jit devtoolset-8-oprofile-devel devtoolset-8-ltrace devtoolset-8-dwz devtoolset-8-make devtoolset-8-memstomp devtoolset-8-strace" && \
+    [ $(uname -m) = "x86_64" ] && INSTALL_PKGS="$INSTALL_PKGS devtoolset-8-dyninst devtoolset-8-dyninst-devel" || : && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    yum -y clean all --enablerepo='*'
```
Comment (phracek) [inline 8-perftools/Dockerfile.rhel7:27]:
```suggestion
    yum clean all -y
```

File: 8-perftools/cccp.yml [hunk 3]
```diff
@@ -0,0 +1 @@
+job-id: devtoolset-7-perftools-centos7
```
Comment (phracek) [inline 8-perftools/cccp.yml:1]: devtoolset-7 is intended? Shouldn't be there `devtoolset-8`?

File: 8-toolchain/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+      description="$DESCRIPTION" \
+      io.k8s.description="$DESCRIPTION" \
+      io.k8s.display-name="Developer Toolset 8 Toolchain" \
+      maintainer="Marek Polacek <polacek@redhat.com>"
+
+RUN yum install -y centos-release-scl-rh && \
+    INSTALL_PKGS="devtoolset-8-gcc devtoolset-8-gcc-c++ devtoolset-8-gcc-gfortran devtoolset-8-gdb" && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    yum -y clean all --enablerepo='*'
```
Comment (phracek) [inline 8-toolchain/Dockerfile:27]:
```suggestion
    yum clean all -y
```

File: 8-toolchain/Dockerfile.fedora [hunk 5]
```diff
@@ -0,0 +1,45 @@
+FROM registry.fedoraproject.org/f27/s2i-core:latest
```
Comment (phracek) [inline 8-toolchain/Dockerfile.fedora:1]: Just a question. Do we really want to support this a bit older image?
What about at least f29? Or even f31 where is a fixed issue https://github.com/sclorg/s2i-base-container/pull/198 although it is not related to C/C++.

File: 8-toolchain/Dockerfile.rhel7 [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+# To use subscription inside container yum command has to be run first (before yum-config-manager)
+# https://access.redhat.com/solutions/1443553
+RUN yum repolist > /dev/null && \
+    yum-config-manager --enable rhel-server-rhscl-7-rpms && \
+    yum-config-manager --enable rhel-7-server-optional-rpms && \
+    yum-config-manager --disable epel >/dev/null || : && \
+    INSTALL_PKGS="devtoolset-8-gcc devtoolset-8-gcc-c++ devtoolset-8-gcc-gfortran devtoolset-8-gdb" && \
+    yum install -y --setopt=tsflags=nodocs $INSTALL_PKGS && \
+    rpm -V $INSTALL_PKGS && \
+    yum -y clean all --enablerepo='*'
```
Comment (phracek) [inline 8-toolchain/Dockerfile.rhel7:32]:
```suggestion
    yum clean all -y
```

**Review by phracek — CHANGES_REQUESTED** [review]
Some changes which need to be fixed. But thanks for the update devtoolset-container.

File: 8-toolchain/cccp.yml [hunk 7]
```diff
@@ -0,0 +1 @@
+job-id: devtoolset-7-toolchain-centos7
```
Comment (phracek) [inline 8-toolchain/cccp.yml:1]:
```suggestion
job-id: devtoolset-8-toolchain-centos7
```

**Review by phracek — CHANGES_REQUESTED** [review]
Please rebase PR.
I  would be glad if you can update also `README.md` file in the top directory.

**Comment by pkubatrh** [comment]
We will no longer be maintaining the image sources upstream for devtoolset-8 and newer images. We are still open for PRs though.

**Pull request closed without merging**